plugins/modules/mongodb_oplog.py

```diff
@@ -182,68 +183,51 @@ def main():
     except Exception as excep:
         module.fail_json(msg='Unable to connect to MongoDB: %s' % to_native(excep))
 
-    if login_user is None and login_password is None:
-        mongocnf_creds = load_mongocnf()
-        if mongocnf_creds is not False:
-            login_user = mongocnf_creds['user']
-            login_password = mongocnf_creds['password']
-    elif login_password is None or login_user is None:
-        module.fail_json(msg="When supplying login arguments, both 'login_user' and 'login_password' must be provided")
+    mongo_auth(module, client)
 
-    if login_user is not None and login_password is not None:
-        try:
-            client.admin.authenticate(login_user, login_password, source=login_database)
-            # Get server version:
-            try:
-                srv_version = LooseVersion(client.server_info()['version'])
-                if srv_version < LooseVersion(ver):
-                    module.fail_json(msg="This module does not support MongoDB {0}".format(srv_version))
-            except Exception as excep:
-                module.fail_json(msg='Unable to get MongoDB server version: %s' % to_native(excep))
-
-            # Get driver version::
-            driver_version = LooseVersion(PyMongoVersion)
-            # Check driver and server version compatibility:
-            check_compatibility(module, srv_version, driver_version)
-        except Exception as excep:
-            module.fail_json(msg='Unable to authenticate with MongoDB: %s' % to_native(excep))
+    try:
+        srv_version = LooseVersion(client.server_info()['version'])
```

Comment:
Can you use mongodb_common.check_srv_version here to get the version?

Reply:
Yes, I actually meant to have done that.